stubs/weak/Reference.php

```diff
@@ -16,13 +16,17 @@
 class Reference
 {
     /**
-     * @param object        $referent Referent object
-     * @param callable|null $notify   Callback to notify when referent object destroyed.
+     * @param object                  $referent Referent object
+     * @param callable | array | null $notify   Optional notifier to signal when referent object destroyed.
      *
-     * Callback will receive current Weak\Reference object as a first argument. Note, that referent object at that time
-     * will be already destroyed.
+     * If notifier is callback, it will be called with a current Weak\Reference object as a first argument. Note, that
+     * referent object at that time will be already destroyed. Callback will not be called if referent object destructor
+     * or previous notify callback throws exception.
+     *
+     * If notifier is array, current Weak\Reference object will be appended to it regardless any exception thrown by
+     * referent object destructor or previous notify callback.
      */
-    public function __construct(object $referent, callable $notify = null)
+    public function __construct(object $referent, $notify = null)
```

Comment:
From the text in the third paragraph, yes, but normally this function-signature will not work, ex:

```PHP
<?php
function addSomething($a){
 $a[] = "Hello world";
}
$a = array();
addSomething($a);
print(count($a)); # Still zero
````

Instead, modern PHP needs the ampersand:

```PHP
function addSomething(&$a){
 $a[] = "Hello world";
}
````